### PR TITLE
Add phase type to claim attributes on calls to claims index

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

- *This work is behind a feature toggle (flipper): YES*
- Adds `phaseType` to the `claimPhaseDates` section for each claim returned from calls to GET /veterans/{veteranId}/claims.
- Corrects existing bugs related to calls to GET /veterans/{{veteran_id}}/claims/{{claim_id}} that were exposed by change to index logic.

## Related issue(s)

[API-37050](https://jira.devops.va.gov/browse/API-37050)

## Testing done

- [ ] *New code is covered by unit tests*
- Tested locally with Postman, confirmed phaseType is displayed as expected as part of the claims show response for each claim.
- Confirmed claim data shared between index and show is displayed the same in either view.

## Screenshots
Screenshot 2024-07-03 at 4 25 11 PM

## What areas of the site does it impact?
Response data returned from calls to claims index.
Logic related to claim phases for both index and show endpoints.

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 1]
```diff
@@ -572,17 +580,15 @@ def build_claim_phase_attributes(bgs_claim, view)
           case view
           when 'show'
             {
-              claim_phase_dates:
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:575]: When returned to `build_claim_structure` on line 186, this resulted in a nested duplicate attribute name in the claim structure (e.g. `claim_phase_dates: { claim_phase_dates: {`). This bug was obscured by the fact that we were calling the hardcoded "index" view previously from that line in `build_claim_structure`.

File: modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb [hunk 2]
```diff
@@ -86,14 +87,14 @@ def generate_show_output(bgs_claim:, lighthouse_claim:) # rubocop:disable Metric
...
-            structure = build_claim_structure(data: bgs_details,
+            structure = build_claim_structure('show',
+                                              data: bgs_details,
                                               lighthouse_id: nil,
                                               upstream_id: bgs_details[:benefit_claim_id])
           end
           structure.merge!(errors: get_errors(lighthouse_claim))
           structure.merge!(supporting_documents: build_supporting_docs(bgs_claim))
           structure.merge!(tracked_items: map_bgs_tracked_items(bgs_claim))
-          structure.merge!(build_claim_phase_attributes(bgs_claim, 'show'))
```
Comment (mchristiansonVA) [inline modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:96]: We were already calling `build_claim_phase_attributes` via `build_claim_structure`, so removed this duplicate call.

Comment (stiehlrod) [reply modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb:96]: @mchristiansonVA Everything looks good! One request: can we add a test to the request spec that uses the [bgs_claim_details_dto_with_phaseback_lc_status](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/spec/factories/auto_established_claims.rb#L189) factory?

**Review by stiehlrod — APPROVED** [review]
LGTM. Tested index and show with various ICNs

**Pull request merged**